lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -51,10 +77,11 @@ def serializable_hash_for_collection(serializer, options)
         end
 
         def serializable_hash_for_single_resource(serializer, options)
+          hash[:data] = {}
           primary_data = primary_data_for(serializer, options)
           relationships = relationships_for(serializer)
           included = included_for(serializer)
-          hash = { data: primary_data }
+          hash[:data].merge!(primary_data)
```

Comment:
C.f. my other comment, I think it should not be. I don't think there's any valid reason to start building the hash in the constructor as it is the sole responsibility of the `serializable_hash` method.